### ISSUE TITLE
Early stopping and/or learning rate decay

### ISSUE THREAD

**Issue by isaacmg** [AIStream-Peelout/flow-forecast#36]
Right now model continues training for set number of epochs even if validation_loss is not improving or getting worse. 
<img width="938" alt="image" src="https://user-images.githubusercontent.com/3865062/71775802-f1e07980-2f54-11ea-80a4-4d5578926f77.png">
This is both a waste of time and could actually be hurting performance. 
- [x] Add param for early stopping in config file where training will terminated after N epochs of non-improvement.
- [ ] Add param for learning rate decay and cyclical learning rate 
- [x]  Unit tests for this function
